action-api/src/main/java/com/powsybl/action/AbstractAction.java

```diff
@@ -25,4 +25,21 @@ protected AbstractAction(String id) {
     public String getId() {
         return id;
     }
+
+    @Override
+    public boolean equals(Object o) {
+        if (this == o) {
+            return true;
+        }
```

Comment:
I think that it would be better to put this check in the non-abstract inheriting classes.

It is meant to avoid unnecessary checks on the attributes if `o` is the same as the current object. But since in the inheriting classes, the `equals` method follows this pattern:
```java
    @Override
    public boolean equals(Object o) {
        if (!super.equals(o)) {
            return false;
        }
        // Checks on the current class' attributes
    }
```
when both objects are the same (hence, `super.equals(o)` will return `true`), every attributes will still be compared.